### PR TITLE
libreoffice-language-pack: install manually

### LINKED ISSUE #52758: libreoffice can not be opened after install
#### General troubleshooting steps

- [x] I have retried my command with `--force` and the issue is still present.
- [x] I have checked the instructions for [reporting bugs](https://github.com/Homebrew/homebrew-cask#reporting-bugs).
- [x] I made doubly sure this is not a [checksum does not match](https://github.com/Homebrew/homebrew-cask/blob/master/doc/reporting_bugs/a_cask_fails_to_install.md#checksum-does-not-match-error) error.
- [x] I ran `brew update-reset && brew update` and retried my command.
- [x] I ran `brew doctor`, fixed as many issues as possible and retried my command.
- [x] I checked there are no [open issues](https://github.com/Homebrew/homebrew-cask/issues) for the same problem.
- [x] I understand that [if I ignore these instructions, my issue may be closed without review](https://github.com/Homebrew/homebrew-cask/blob/master/doc/faq/closing_issues_without_review.md).

#### Description of issue

LibreOffice can not be opened after install. It opens a dialog which suggests to move the app to trash. Even `brew cask zap libreoffice libreoffice-language-pack` and reinstalling afterwards does not help.

Installing LibreOffice with the current version (6.1.2) from the LibreOffice website incl. language pack works fine.

Thanks for fixing in advance.

#### Output of `brew cask doctor`

```
==> Homebrew Version
1.7.6-66-g8f74ba0
==> macOS
10.14
==> SIP
Disabled
==> Java
11
==> Gatekeeper support
Enabled
==> Homebrew Cask Install Location
<NONE>
==> Homebrew Cask Staging Location
/usr/local/Caskroom
==> Homebrew Cask Taps:
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask (4036 casks)
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask-versions (204 casks)
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask-eid (11 casks)
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask-fonts (1177 casks)
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask-drivers (152 casks)
==> Contents of $LOAD_PATH
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/ruby-macho-2.1.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/plist-3.4.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/backports-3.11.4/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/activesupport-5.2.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/tzinfo-1.2.5/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/thread_safe-0.3.6/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/minitest-5.11.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/i18n-1.1.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/concurrent-ruby-1.0.5/lib
/Library/Ruby/Gems/2.3.0/gems/did_you_mean-1.0.0/lib
/Library/Ruby/Site/2.3.0
/Library/Ruby/Site/2.3.0/x86_64-darwin18
/Library/Ruby/Site/2.3.0/universal-darwin18
/Library/Ruby/Site
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/vendor_ruby/2.3.0
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/vendor_ruby/2.3.0/x86_64-darwin18
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/vendor_ruby/2.3.0/universal-darwin18
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/vendor_ruby
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/x86_64-darwin18
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/universal-darwin18
/usr/local/Homebrew/Library/Homebrew
==> Environment Variables
LC_ALL="en_US.UTF-8"
PATH="/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/Homebrew/Library/Homebrew/shims/scm"
SHELL="/bin/bash"
```

#### Output of `brew tap`

```
homebrew/cask
homebrew/cask-drivers
homebrew/cask-eid
homebrew/cask-fonts
homebrew/cask-versions
homebrew/core
mas-cli/tap
vitorgalvao/tiny-scripts
```

### PR DESCRIPTION
**Pull request by amyspark**
After making all changes to the cask:

- [X] `brew cask audit --download {{cask_file}}` is error-free.
- [X] `brew cask style --fix {{cask_file}}` reports no offenses.
- [X] The commit message includes the cask’s name and version.
- [X] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, if **adding a new cask**:

- [ ] Named the cask according to the [token reference].
- [ ] `brew cask install {{cask_file}}` worked successfully.
- [ ] `brew cask uninstall {{cask_file}}` worked successfully.
- [ ] Checked there are no [open pull requests] for the same cask.
- [ ] Checked the cask was not [already refused].
- [ ] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/Homebrew/homebrew-cask/pulls
[already refused]: https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues
[the correct repo]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask
[version-checksum]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256

--

LibreOffice Language Pack, when manually executed, runs LibreOffice at least once (using AppleScript) before inserting the language files. This is to trigger Gatekeeper and thus cheat the code signature assessment.

This pull request ~~adds this step to the `preflight` block.~~ changes this Cask to manual installation.

Fixes #52758 , fixes #52883 .

~~(source for the fix: `Contents/osx_install.applescript` inside the Cask's DMG)~~

### CODE REVIEW

**Review by vitorgalvao — CHANGES_REQUESTED** [review]
Lets just add the caveat. Adding the AppleScript is too much, and HBC should never trigger GUIs, especially one as taxing as LibreOffice.

**Comment by amyspark** [comment]
I understand @vitorgalvao. But since this step is essential for LibreOffice to work out of the box (TDF even adds it in their own executable), I'd like to get a consensus on this. 
Pinging @Homebrew/cask.

**Comment by commitay** [comment]
> Adding the AppleScript is too much

👍 

> and HBC should never trigger GUIs

👍 

> I’d rather remove the auto-installation (which is a non-standard hacky method anyway) and change the cask to `installer :manual`.

👍

**Comment by tiiiecherle** [comment]
I know you guys come across a lot of issues and have standards to solve them, but just from a user and automation point of view adding the applescript seems the best way to implement. As using the homebrew-cask install one of the most things I love about it is the automation. And if there is a way to do the install automatically without breaking downsides I (and I guess other users) would love to see a most effortless install as possible. For me as an example I would have to explain to everyone in the office what to do and why instead of just installing the cask.

Would be nice to have see the most automated version possible. Thanks a lot for all your work. I really do like this project. Have a nice weekend together ;)

**Comment by vitorgalvao** [comment]
> but just from a user and automation point of view adding the applescript seems the best way to implement.

Unfortunately, that doesn’t consider the users who don’t want a GUI to spring up when installing, especially one that’s slow like LibreOffice.

**Comment by tiiiecherle** [comment]
I am pretty sure there is no "real" popup of a window when install the language pack manually. I will try and report back in a few minutes...

**Comment by tiiiecherle** [comment]
I just downloaded the latest libreoffice and language pack from the offical website. Then I dragged the LibreOffice.app to /Applications/LibreOffice.app and unmounted the .dmg.
Afterwards I mounted the Language Pack installer and double clicked it. You are right, LibreOffice opens, but only for a very short time, not really noticable and is quit really fast before showing the startup gui. Then then langpack installs and reports success.

The only thing is that libreoffice asks for permission to open on first launch due to gatekeeper. This could be avoided by exporting the quarantine data, deleting the xattr, open Libreoffice and reimport the quarantine afterwards like this:
```
xattr -p com.apple.quarantine "/Applications/LibreOffice.app" > /tmp/quarantine_libreoffice.xattr
xattr -d com.apple.quarantine "/Applications/LibreOffice.app"
INSTALL LANGUAGE PACK HERE
xattr -w com.apple.quarantine `cat "/tmp/quarantine_libreoffice.xattr"` "/Applications/LibreOffice.app"

```

**Comment by tiiiecherle** [comment]
I tried that, unfortunately restoring the com.apple.quarantine xattr does seem to break libreoffice again after the process completes. So it would still be possible with the applescriop, the only thing a user would have to do is hit one button to allow libreoffice to be opened, the rest could be automated.

**Comment by amyspark** [comment]
> I tried that, unfortunately restoring the com.apple.quarantine xattr does seem to break libreoffice again after the process completes. So it would still be possible with the applescriop, the only thing a user would have to do is hit one button to allow libreoffice to be opened, the rest could be automated without long waiting and without seeing any libreoffice gui.

What you actually want is to toggle the "opened" bit in `com.apple.quarantine` - but doing that without updating the security policy database is... prone to errors 😉

File: Casks/libreoffice-language-pack.rb [hunk 1]
```diff
@@ -564,12 +564,7 @@
...
-  stage_only true
-
-  preflight do
-    system_command '/usr/bin/tar', args: ['-C', "#{appdir}/LibreOffice.app/", '-xjf', "#{staged_path}/LibreOffice Language Pack.app/Contents/tarball.tar.bz2"]
-    system_command '/usr/bin/touch', args: ["#{appdir}/LibreOffice.app/Contents/Resources/extensions"]
-  end
+  installer manual: 'LibreOffice Language Pack.app'
 
   caveats <<~EOS
     #{token} assumes LibreOffice is installed in '#{appdir}'. If it is not, you’ll need to run '#{staged_path}/LibreOffice Language Pack.app' manually.
```
Comment (commitay) [inline Casks/libreoffice-language-pack.rb]: The `caveat` isn't needed now with the switch to `installer manual`

**Comment by commitay** [comment]
```
audit for libreoffice-language-pack: warning
 - installer and pkg stanzas require an uninstall stanza
```
Need to add a fake `uninstall` to make `audit` happy.
https://github.com/Homebrew/homebrew-cask/blob/959f4d949808c895fbc507e097936779b314a208/Casks/google-cloud-sdk.rb#L19

**Review by vitorgalvao — APPROVED** [review]

**Pull request merged**